WcaOnRails/app/models/user.rb

```diff
@@ -666,7 +672,11 @@ def can_admin_competitions?
   alias_method :can_announce_competitions?, :can_admin_competitions?
 
   def can_manage_competition?(competition)
-    can_admin_competitions? || competition.organizers.include?(self) || competition.delegates.include?(self) || wrc_team? || competition.delegates.map(&:senior_delegate).compact.include?(self) || ethics_committee?
+    (
+      can_admin_competitions? || competition.organizers.include?(self) ||
+      competition.delegates.include?(self) || competition.trainee_delegates.include?(self) ||
+      wrc_team? || competition.delegates.map(&:senior_delegate).compact.include?(self) || ethics_committee?
```

Comment:
Very nitpicky comment: given we can't put all on the same line, it may be a good idea to put one per line; this makes it clear it's just an enumeration of all the people allowed to manage the competition.